aws/resource_aws_organizations_account.go

```diff
@@ -0,0 +1,228 @@
+	conn := meta.(*AWSClient).organizationsconn
+
+	// Create the account
+	createOpts := &organizations.CreateAccountInput{
+		AccountName: aws.String(d.Get("name").(string)),
+		Email:       aws.String(d.Get("email").(string)),
+	}
+	if role, ok := d.GetOk("role_name"); ok {
+		createOpts.RoleName = aws.String(role.(string))
+	}
+
+	if iam_user, ok := d.GetOk("iam_user_access_to_billing"); ok {
+		createOpts.IamUserAccessToBilling = aws.String(iam_user.(string))
+	}
+
+	log.Printf("[DEBUG] Account create config: %#v", createOpts)
+
+	var err error
+	var resp *organizations.CreateAccountOutput
+	err = resource.Retry(4*time.Minute, func() *resource.RetryError {
+		resp, err = conn.CreateAccount(createOpts)
+
+		if err != nil {
+			ec2err, ok := err.(awserr.Error)
+			if !ok {
+				return resource.NonRetryableError(err)
+			}
+			if isAWSErr(ec2err, organizations.ErrCodeFinalizingOrganizationException, "") {
+				log.Printf("[DEBUG] Trying to create account again: %q", ec2err.Message())
+				return resource.RetryableError(err)
+			}
+
+			return resource.NonRetryableError(err)
+		}
+
+		return nil
+	})
+
+	if err != nil {
+		return fmt.Errorf("Error creating account: %s", err)
+	}
+	log.Printf("[DEBUG] Account create response: %#v", resp)
+
+	requestId := *resp.CreateAccountStatus.Id
+
+	// Wait for the account to become available
+	log.Printf("[DEBUG] Waiting for account request (%s) to succeed", requestId)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:      []string{organizations.CreateAccountStateInProgress},
+		Target:       []string{organizations.CreateAccountStateSucceeded},
+		Refresh:      resourceAwsOrganizationsAccountStateRefreshFunc(conn, requestId),
+		PollInterval: 10 * time.Second,
+		Timeout:      5 * time.Minute,
+	}
+	stateResp, stateErr := stateConf.WaitForState()
+	if stateErr != nil {
+		return fmt.Errorf(
+			"Error waiting for account request (%s) to become available: %s",
+			requestId, stateErr)
+	}
+
+	// Store the ID
+	accountId := stateResp.(*organizations.CreateAccountStatus).AccountId
+	d.SetId(*accountId)
+
+	return resourceAwsOrganizationsAccountRead(d, meta)
+}
+
+func resourceAwsOrganizationsAccountRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).organizationsconn
+	describeOpts := &organizations.DescribeAccountInput{
+		AccountId: aws.String(d.Id()),
+	}
+	resp, err := conn.DescribeAccount(describeOpts)
+	if err != nil {
+		if isAWSErr(err, organizations.ErrCodeAccountNotFoundException, "") {
+			log.Printf("[WARN] Account does not exist, removing from state: %s", d.Id())
+			d.SetId("")
+			return nil
+		}
+		return err
+	}
+
+	d.Set("arn", resp.Account.Arn)
+	d.Set("joined_method", resp.Account.JoinedMethod)
+	d.Set("joined_timestamp", resp.Account.JoinedTimestamp)
+	d.Set("name", resp.Account.Name)
+	d.Set("status", resp.Account.Status)
+	return nil
+}
+
+func resourceAwsOrganizationsAccountDelete(d *schema.ResourceData, meta interface{}) error {
```

Comment:
Due to how the AWS Organizations service and closing AWS accounts in general works, we should probably set up the Terraform deletion function here to call [`RemoveAccountFromOrganization`](https://docs.aws.amazon.com/sdk-for-go/api/service/organizations/#Organizations.RemoveAccountFromOrganization) and pass along the error to the Terraform user. I put a review comment to update the documentation appropriately as well. e.g.

```go
func resourceAwsOrganizationsAccountDelete(d *schema.ResourceData, meta interface{}) error {
  conn := meta.(*AWSClient).organizationsconn

  input := &organizations.RemoveAccountFromOrganizationInput{
    AccountId: aws.String(d.Id()),
  }
  log.Printf("[DEBUG] Removing AWS account from organization: %s", input)
  _, err := conn.RemoveAccountFromOrganization(input)
  if err != nil {
    if isAWSErr(err, organizations.ErrCodeAccountNotFoundException, "") {
      return nil
    }
    return err
  }
  return nil
}
```

Reply:
OK, I can do that.  In fact, since you wrote the code feel free to update the PR.  Let me know either way, thanks.